### ISSUE TITLE
Embedding layer - add default behavior for out of bound index to documentation

### ISSUE THREAD

**Issue by bejjani** [apache/incubator-mxnet#8940]
https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/indexing_op.h#L613

The current default/only behavior of the embedding layer in case of an out of bound index is to "clip" to max index of the embedding layer.
The documentation does not mention it.

Another suggestion is to add the option to assign a random embedding during prediction for out of bound indexes. That makes more sense to me compared to clipping.

**Comment by piiswrong** [comment]
@eric-haibin-lin

**Comment by eric-haibin-lin** [comment]
It was mentioned in the doc (but maybe not obvious since there are a lot of texts)
https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html#mxnet.ndarray.Embedding

`contrib.SparseEmbedding` op does throw an exception on out of bound error
Suggestion for doc improvement is welcome

**Comment by bejjani** [comment]
You're right, doc for ndarray.Embedding does mention however "By default, if any index mentioned is too large, it is replaced by the index that addresses the last vector in an embedding matrix."
It seems to suggest that other options are available.

So the sparse implementation behaves differently and throws an exception instead?

**Comment by eric-haibin-lin** [comment]
Yes the sparse implementation will throw an exception instead. 
Is anyone else using random assignment on out-of-bound indices? If the use case makes sense, an optional param can be added to Embedding to override the default behavior.

**Comment by leleamol** [comment]
Proposed Labels: "Feature Request","Discussion","Unclear Error/Doc","Sparse","Exception Handling"
